src/commands/auth/logout.ts

```diff
@@ -36,20 +38,46 @@ export default class Logout extends SfdxCommand {
 
   public async run(): Promise<string[]> {
     if (this.flags.targetusername && this.flags.all) {
-      const err = new SfdxError(messages.getMessage('specifiedBothUserAndAllError'), 'SpecifiedBothUserAndAllError');
-      return Promise.reject(err);
+      throw new SfError(messages.getMessage('specifiedBothUserAndAllError'), 'SpecifiedBothUserAndAllError');
     }
 
     const remover = await AuthRemover.create();
+    let usernames: string[];
+    let usingSfdxFolder = false;
 
-    const authConfigs = this.shouldFindAllAuths()
-      ? await remover.findAllAuthConfigs()
-      : await remover.findAuthConfigs(this.flags.targetusername);
+    const ca = ConfigAggregator.getInstance();
 
-    if (await this.shouldRunCommand(authConfigs)) {
-      const usernames = [...authConfigs.keys()];
+    const targetUsername = this.flags.targetusername
+      ? (this.flags.targetusername as string)
+      : (ca.getInfo(OrgConfigProperties.TARGET_ORG).value as string);
+
+    try {
+      usernames = this.shouldFindAllAuths()
+        ? Object.keys(remover.findAllAuths())
+        : [(await remover.findAuth(targetUsername)).username];
+    } catch (e) {
+      if ((e as SfError).name === 'NamedOrgNotFound') {
+        // try looking in the old ~/.sfdx folder
+        // TODO: Can be removed once everything is in the `~/.sf` folder
+        if (fs.existsSync(join('~', '.sfdx', `${targetUsername}.json`))) {
+          usingSfdxFolder = true;
+          usernames = [targetUsername];
+        }
+      } else {
+        // can't type e as Error. catch(e) requires e as unknown or any
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+        e.name = 'NoOrgFound';
+        throw e;
+      }
+    }
+
+    if (await this.shouldRunCommand(usernames)) {
       for (const username of usernames) {
-        await remover.removeAuth(username);
+        if (usingSfdxFolder) {
+          await fs.promises.rm(join('~', '.sfdx', `${targetUsername}.json`));
+        } else {
+          await remover.removeAuth(username);
+        }
```

Comment:
The auth remover functions take care of resolving aliases and fetching `target-org` from config, as well as deleting the same on remove. So I believe the following code is all that is needed.

The error handling for org not found should not be needed, as global info should be handling keeping sf and sfdx auth's and configs in balance.
``` suggestion

  const usernames = this.flags.all ? remover.findAllAuths() : [remover.findAuth(this.flags.targetusername)];

  if (await this.shouldRunCommand(usernames)) {
      for (const username of usernames) {
          await remover.removeAuth(username);
      }
```